ebos/ecltransmissibility_impl.hh

```diff
@@ -866,13 +874,40 @@ applyNncToGridTrans_(const std::unordered_map<std::size_t,int>& cartesianToCompr
             continue;
         }
 
-        auto candidate = trans_.find(isId(low, high));
-        if (candidate != trans_.end()) {
-            // NNC is represented by the grid and might be a neighboring connection
-            // In this case the transmissibilty is added to the value already
-            // set or computed.
-            candidate->second += nncEntry.trans;
+        {
+            auto candidate = trans_.find(isId(low, high));
+            if (candidate != trans_.end()) {
+                // NNC is represented by the grid and might be a neighboring connection
+                // In this case the transmissibilty is added to the value already
+                // set or computed.
+                candidate->second += nncEntry.trans;
+            }
         }
+        // if (enableEnergy_) {
+        //     auto candidate = thermalHalfTrans_.find(directionalIsId(low, high));
+        //     if (candidate != trans_.end()) {
+        //         // NNC is represented by the grid and might be a neighboring connection
+        //         // In this case the transmissibilty is added to the value already
+        //         // set or computed.
+        //         candidate->second += nncEntry.transEnergy1;
+        //     }
+        //     auto candidate = thermalHalfTrans_.find(directionalIsId(high, low));
+        //     if (candidate != trans_.end()) {
+        //         // NNC is represented by the grid and might be a neighboring connection
+        //         // In this case the transmissibilty is added to the value already
+        //         // set or computed.
+        //         candidate->second += nncEntry.transEnergy2;
+        //     }
+        // }
+        // if (enableDiffusivity_) {
+        //     auto candidate = diffusivity_.find(isId(low, high));
+        //     if (candidate != trans_.end()) {
+        //         // NNC is represented by the grid and might be a neighboring connection
+        //         // In this case the transmissibilty is added to the value already
+        //         // set or computed.
+        //         candidate->second += nncEntry.transDiffusion;
+        //     }
+        // }
     }
```

Comment:
maybe remove this commented code to clean up?

Reply:
I just added it to show where to ad properties if one need it for diffusion and energy and it is added in input + nncEntry.

Reply:
I see.
Indeed, NNC's are treated differently in OPM depending on their direction. In Z-direction (connections over pinched out cells) NNCs are repesented as normal faces. In this case Energy and Diffusion will be computed for them. For all explicitly specified NNCs this is not the case and we would need an additional custom keyword.